packages/loader/container-loader/src/container.ts

```diff
@@ -801,6 +810,40 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
             this.emit("closed", error);
 
+            if (this.visibilityEventHandler !== undefined) {
+                document.removeEventListener("visibilitychange", this.visibilityEventHandler);
+            }
+        } finally {
+            this._lifecycleState = "closed";
```

Comment:
Seems worth adding a new lifecycleState for "disposed", which is after closed and basically equivalent to it (but it's logged so will be interesting to see the difference).  Basically equivalent except in these functions, I guess.